ui/packages/consul-ui/app/components/topology-metrics/index.js

```diff
@@ -66,21 +69,65 @@ export default class TopologyMetrics extends Component {
       });
   }
 
+  emptyColumn() {
+    const noDependencies = get(this.args.topology, 'noDependecies');
+    return !this.env.var('CONSUL_ACLS_ENABLED') || noDependencies;
+  }
+
+  get downstreams() {
+    const downstreams = get(this.args.topology, 'Downstreams') || [];
+    const items = [...downstreams];
+    const noDependencies = get(this.args.topology, 'noDependecies');
+
+    if (!this.env.var('CONSUL_ACLS_ENABLED')) {
```

Comment:
Looking at the existing test that uses this functionality and the step itself, the step seems to act on HTTP requests only. It was originally used to mock out a HTTP API with disabled ACLs (in order to respond with a 401 status) so that makes sense also.

It looks like here you need to mock out settings during Ember's bootup process, basically mock out the settings we add into the `index.html` file. Ember's test runner instantiates with ACLs enabled for the frontend it seems and once set that cannot be changed because during testing Ember doesn't refresh the page like proper e2e testing would.

So if we've hit the point where we need to test this case we'll have to find a way to change (or appear to change) 'bootup settings' for every single test.

Reply:
@johncowen I tried to mock settings like this: 

```
When settings from yaml
---
  CONSUL_ACLS_ENABLE: 0
---
```

It did not work. Is there a specific way to format mocking settings?

Reply:
There is no way currently to turn off ACLs for the frontend once the application has booted up. You can only turn off ACLs (well, better said 'toggle') for the HTTP API once the frontend has booted. I believe all the pieces are there to be able to do this 'fake' toggling bootup config, but it's not all joined up as yet.

You won't be able to write the specific acceptance test you are writing for this without adding a bunch of extra app/test code.

---

`Given settings from yaml` mocks out the users settings (in localStorage) for example the users token, not operator/bootup config. This _is_ currently mockable after bootup.

Reply:
It sounds like we don't have a good/solid way to test this reliably at the moment (unfortunately) so for now we likely need to skip until we can find a solution.